### PR TITLE
Base Modal

### PR DESCRIPTION
**Pull request by smyth64**
Included:
- Modal
- UnitTests
- Storybook

image

### CODE REVIEW

File: src/components/ui/BaseButton.vue [hunk 1]
```diff
@@ -26,5 +26,11 @@ export default {
 	&:focus {
 		background-color: transparentize($color-text, 0.1);
 	}
+	.is-light {
+		color: #363636;
+		background-color: #fff;
+		border-color: #dbdbdb;
```
Comment (adrianjost) [inline src/components/ui/BaseButton.vue]: @domi should we use variables here instead?

File: src/components/ui/BaseModal.unit.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			<BaseButton id="button" class="is-light" @click="showModal = false">
+				OK
+			</BaseButton>
+		</div>
+	</BaseModal>				
+	`,
+	components: { BaseModal },
+};
+
+describe("@components/BaseButton", () => {
```
Comment (adrianjost) [inline src/components/ui/BaseModal.unit.js]: `BaseButton` ?

File: src/components/ui/BaseModal.vue [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		<div v-if="showModal" class="modal-mask">
+			<div
+				class="modal-wrapper"
+				@mousedown.self="handleBackgroundClick"
+				@touchstart.self="handleBackgroundClick"
+			>
+				<div class="modal-container">
+					<div class="modal-header">
+						<slot name="header">
+							default header
```
Comment (adrianjost) [inline src/components/ui/BaseModal.vue]: I would use HTML-comments for the default values or remove them entirely so you don't need to specify something empty when you don't want to use this slot at all.

File: src/components/ui/BaseModal.vue [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+		showModal: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	methods: {
+		handleBackgroundClick() {
+			this.toggle();
+		},
+		toggle() {
```
Comment (adrianjost) [inline src/components/ui/BaseModal.vue]: If toggle is always emitting `close`, wouldn't it be better to name it `close`?

File: src/components/ui/BaseModal.vue [hunk 5]
```diff
@@ -0,0 +1,143 @@
+<template>
+	<transition name="modal">
+		<div v-if="showModal" class="modal-mask">
+			<div
+				class="modal-wrapper"
+				@mousedown.self="handleBackgroundClick"
+				@touchstart.self="handleBackgroundClick"
+			>
+				<div class="modal-container">
+					<div class="modal-header">
```
Comment (adrianjost) [inline src/components/ui/BaseModal.vue]: Idea to make this component even more generic: 
Use a nested slot structure like this:
```
div.modal-container
  slot
    div.modal-header
      slot name=header
    div.modal-body
      slot name=body
    ...
```
Then you can use the component even for modals with completely customized layouts as a (like the name hints us) Base Component

File: src/components/ui/BaseButton.vue (hunk 1 above)
Comment (dominikbruechner) [reply src/components/ui/BaseButton.vue]: Yes, lightColor, or if not generic buttonLightColor. ansonsten colorBlack für fff

File: src/components/ui/BaseModal.vue (hunk 5 above)
Comment (smyth64) [reply src/components/ui/BaseModal.vue]: Good idea, gonna implement it.

File: src/components/ui/BaseModal.vue (hunk 4 above)
Comment (smyth64) [reply src/components/ui/BaseModal.vue]: Gonna change it, also the name for showModal to -> active.

File: src/components/ui/BaseButton.vue (hunk 1 above)
Comment (smyth64) [reply src/components/ui/BaseButton.vue]: ok i will set this variable into the default theme.

File: src/components/ui/BaseModal.vue (hunk 3 above)
Comment (smyth64) [reply src/components/ui/BaseModal.vue]: alright

File: src/components/ui/BaseModal.vue (hunk 5 above)
Comment (smyth64) [reply src/components/ui/BaseModal.vue]: unfortunately slot in slot is not possible, so we have to use just slot instead

**Comment by adrianjost** [comment]
@smyth64 You don't need to import BaseComponents. They are always available. I changed that for you. (https://github.com/schul-cloud/nuxt-client/pull/63/commits/d8e5635e8d9b8616bddd56978a55ec53af67dfc6) 

Do you have any idea, why the BaseModal isn't working in Storybook? 
At `/news/_id/edit` it's working perfectly for me but using storybook I can't open it at all. Any ideas?

File: src/pages/news/_id/edit.vue [hunk 6]
```diff
@@ -44,22 +77,22 @@ export default {
...
-					try {
-						await this.remove(this.news._id);
-						this.$toast.success("Artikel gelöscht");
-						this.$router.push({ name: "news" });
-					} catch (e) {
-						this.$toast.error("Fehler beim Löschen");
-					}
-				},
-			});
+			// this.$dialog.confirm({
```
Comment (adrianjost) [inline src/pages/news/_id/edit.vue]: this comment should be removed

File: src/pages/news/_id/edit.vue [hunk 7]
```diff
@@ -1,38 +1,71 @@
...
+				<h1>{{ news.title }}</h1>
+				<!-- eslint-disable-next-line vue/no-v-html -->
+				<div v-html="news.content" />
+			</section>
+		</div>
+
+		<!-- use the modal component, pass in the prop -->
+		<BaseModal ref="modal" :active.sync="active">
+			<div class="modal-header">
+				<h3>custom header</h3>
```
Comment (adrianjost) [inline src/pages/news/_id/edit.vue]: 👆

File: src/pages/news/_id/edit.vue (hunk 6 above)
Comment (smyth64) [reply src/pages/news/_id/edit.vue]: ja der kann raus

File: src/pages/news/_id/edit.vue (hunk 7 above)
Comment (smyth64) [reply src/pages/news/_id/edit.vue]: if you talk about the comment, I removed it ;)

**Comment by smyth64** [comment]
> @smyth64 You don't need to import BaseComponents. They are always available. I changed that for you. ([d8e5635](https://github.com/schul-cloud/nuxt-client/commit/d8e5635e8d9b8616bddd56978a55ec53af67dfc6))
> 
> Do you have any idea, why the BaseModal isn't working in Storybook?
> At `/news/_id/edit` it's working perfectly for me but using storybook I can't open it at all. Any ideas?

Very nice, thanks.
I was wondering before why we don't make these components globally available.

Perfect! 👍

**Comment by adrianjost** [comment]
> @smyth64 You don't need to import BaseComponents. They are always available. I changed that for you. ([d8e5635](https://github.com/schul-cloud/nuxt-client/commit/d8e5635e8d9b8616bddd56978a55ec53af67dfc6))
> 
> Do you have any idea, why the BaseModal isn't working in Storybook?
> At `/news/_id/edit` it's working perfectly for me but using storybook I can't open it at all. Any ideas?

Storybook is now working

File: src/pages/news/_id/edit.vue (hunk 7 above)
Comment (adrianjost) [reply src/pages/news/_id/edit.vue]: I talked about the title `custom header`. I changed that now

**Pull request merged**